packages/account/src/Sections/Profile/LanguageSettings/language-settings.tsx

```diff
@@ -1,16 +1,23 @@
 import React from 'react';
 import { Redirect } from 'react-router-dom';
 import { localize, getAllowedLanguages } from '@deriv/translations';
-import { isMobile, routes } from '@deriv/shared';
+import { useFeatureFlags, useStoreWalletAccountsList } from '@deriv/hooks';
+import { routes } from '@deriv/shared';
 import { observer, useStore } from '@deriv/stores';
 import FormSubHeader from 'Components/form-sub-header';
 import LanguageRadioButton from 'Components/language-settings';
 
 const LanguageSettings = observer(() => {
-    const { common } = useStore();
+    const { common, ui } = useStore();
+    const { is_mobile } = ui;
     const { changeSelectedLanguage, current_language } = common;
+    const { has_wallet } = useStoreWalletAccountsList();
+    const { is_next_wallet_enabled } = useFeatureFlags();
 
-    if (window.location.pathname === routes.languages && isMobile()) {
```

Comment:
we can remove `window.location.pathname === routes.languages` because `LanguageSettings` is only rendering when the route is equals to `/account/languages`.